core/storage/couchStorage.go

```diff
@@ -0,0 +1,691 @@
+	// 		tlsConfig.InsecureSkipVerify = true
+	// 	}
+
+	// 	setXport := couchdb.SetTransport(&http.Transport{TLSClientConfig: tlsConfig})
+	// 	client, err = kivik.New("couch", store.dsn)
+	// 	if err != nil {
+	// 		message := fmt.Sprintf("Failed to connect. Error: %s.", err)
+	// 		return &Error{message}
+	// 	}
+
+	// 	err = client.Authenticate(context.TODO(), setXport)
+	// 	if err != nil {
+	// 		message := fmt.Sprintf("Authentication Failed. Error: %s.", err)
+	// 		return &Error{message}
+	// 	}
+	//}
+
+	// basicAuth := couchdb.BasicAuth(store.loginInfo["Username"], store.loginInfo["Password"])
+	// err = client.Authenticate(context.TODO(), basicAuth)
+	// if err != nil {
+	// 	return err
+	// }
+
+	client, err = kivik.New("couch", store.dsn)
+	if client == nil || err != nil {
+		message := fmt.Sprintf("Failed to connect to couch. Error: %s.", err)
+		return &Error{message}
+	}
+
+	exists, err := client.DBExists(context.TODO(), store.loginInfo["dbName"])
+	if !exists {
+		client.CreateDB(context.TODO(), store.loginInfo["dbName"])
+	}
+
+	store.connected = true
+	store.client = client
+
+	return nil
+}
+
+// Stop stops the CouchStorage store
+func (store *CouchStorage) Stop() {
+	store.client.Close(context.TODO())
+}
+
+// PerformMaintenance performs store's maintenance
+func (store *CouchStorage) PerformMaintenance() {
+	//store.checkObjects()
+}
+
+// Cleanup erase the on disk Bolt database only for ESS and test
+func (store *CouchStorage) Cleanup(isTest bool) common.SyncServiceError {
+	return nil
+}
+
+// GetObjectsToActivate returns inactive objects that are ready to be activated
+func (store *CouchStorage) GetObjectsToActivate() ([]common.MetaData, common.SyncServiceError) {
+	return nil, nil
+}
+
+// StoreObject stores an object
+// If the object already exists, return the changes in its destinations list (for CSS) - return the list of deleted destinations
+func (store *CouchStorage) StoreObject(metaData common.MetaData, data []byte, status string) ([]common.StoreDestinationStatus, common.SyncServiceError) {
+	id := getObjectCollectionID(metaData)
+	var dests []common.StoreDestinationStatus
+	var deletedDests []common.StoreDestinationStatus
+
+	if metaData.DestinationPolicy != nil {
+		metaData.DestinationPolicy.Timestamp = time.Now().UTC().UnixNano()
+	}
+
+	if status == common.NotReadyToSend || status == common.ReadyToSend {
+		// The object was receieved from a service, i.e. this node is the origin of the object:
+		// set its instance id and create destinations array
+		newID := store.getInstanceID()
+		metaData.InstanceID = newID
+		if data != nil && !metaData.NoData && !metaData.MetaOnly {
+			metaData.DataID = newID
+		}
+
+		var err error
+		dests, deletedDests, err = createDestinationsFromMeta(store, metaData)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	existingObject := &couchObject{}
+	if err := store.getOne(id, existingObject); err != nil {
+		if !strings.HasPrefix(err.Error(), "Not Found") {
```

Comment:
This is not the best way to check for errors. Check for the common.NotFound error type.

Reply:
Yes, this is directly related to the previous comments above..